docs/design/2018-10-24-view-support.md

```diff
@@ -0,0 +1,49 @@
+# Proposal: Implement View Feature
+
+- Author(s):     AndrewDi
+- Last updated:  2018-10-24
+- Discussion at: https://github.com/pingcap/tidb/issues/7974
+
+## Abstract
+This proposal proposes to implement view feature in TiDB, aimed to make SQL more easy to write.
+
+## Background
+A database view is a searchable object in a database that is defined by a query. Though a view doesn’t store data, some refer to a view as “virtual tables,” you can query a view like you can query a table. A view can combine data from two or more tables, using joins, and also just contain a subset of information. This makes them convenient to abstract, or hide, complicated queries.
+
+Below is a visual depiction of a view:
+  ![AnatomyOfAview](.imgs/view.png)
+  
+A view is created from a query using the CREATE OR REPLACE VIEW command. In the example below we are creating a PopularBooks view based on a query which selects all Books that have the IsPopular field checked. Following is the query:
+`CREATE OR REPLACE VIEW PopularBooks AS SELECT ISBN, Title, Author,PublishDate FROM Books WHERE IsPopular=1`  
+
+Once a view is created, you can use it as you query any table in a SELECT statement. For example, to list all the popular book titles ordered by an author, you could write:  
+`SELECT Author, Title FROM PopularBooks ORDER BY Author`  
+
+In general you can use any of the SELECT clauses, such as GROUP BY, in a select statement containing a view.
+
+## Proposal
+I implement the `create or replace` command to create a view with the SELECT statement. Currently, I reuse `drop table` to drop a view, and `show full tables` can also display the view list with different table types. The `show create table` command can also regenerate a view create command.
```

Comment:
1. This is the restrictions on the `View` in MySQL 5.7, which should be considered as well: https://dev.mysql.com/doc/refman/5.7/en/view-restrictions.html

2. Please describe in detail that how to process a query on a View. 

3. I noticed that you store the original text of the view, maybe it's not proper, because in MySQL: 
	> The view definition is “frozen” at creation time and is not affected by subsequent changes to the definitions of the underlying tables. For example, if a view is defined as SELECT * on a table, new columns added to the table later do not become part of the view, and columns dropped from the table will result in an error when selecting from the view.

	ref: https://dev.mysql.com/doc/refman/5.7/en/create-view.html

Reply:
And as you can see in the view creation, we should also check the privileges. This part should also be described in detail.

ref: https://dev.mysql.com/doc/refman/5.7/en/create-view.html

Reply:
We should also store the view definition into the `INFORMATION_SCHEMA.VIEW` table, please take a look at this: https://dev.mysql.com/doc/refman/5.7/en/views-table.html document.

Reply:
Also, the updatable and insertable views should also be considered: https://dev.mysql.com/doc/refman/5.7/en/view-updatability.html

Reply:
@zz-jason add more detail into document.